jinahub/indexers/searcher/HnswlibSearcher/tests/integration/test_searcher.py

```diff
@@ -0,0 +1,90 @@
+import numpy as np
+import pytest
+from hnswlib_searcher import HnswlibSearcher
+from jina import Document, DocumentArray, Flow
+
+_DIM = 10
+
+
+@pytest.mark.parametrize('uses', ['HnswlibSearcher', 'docker://hnswlibsearcher'])
+def test_index_search_flow(uses: str, build_docker_image: str):
+    f = Flow().add(uses=uses, uses_with={'metric': 'l2', 'dim': _DIM})
+    da = DocumentArray(
+        [
+            Document(id='a', embedding=np.ones(_DIM) * 1.0),
+            Document(id='b', embedding=np.ones(_DIM) * 2.0),
+        ]
+    )
+
+    with f:
+        f.index(da)
+
+        status_ind = f.post('/status', return_results=True)
+        status_ind = status_ind[0].data.docs[0].tags
+
+        assert status_ind['current_indexed'] == 2
+        assert status_ind['total_deleted'] == 0
+        assert status_ind['total_indexed'] == 2
+
+        result_search = f.search(da, return_results=True)
+        result_search = result_search[0].data.docs
```

Comment:
assert size of results. Else the `for` might be skipped if no results and the test will still pass.